CONTRIBUTING.md

```diff
@@ -10,81 +11,200 @@ All work on Illa-builder happens directly on GitHub. Both core team members and
+You can self-host locally with ILLA CLI, Docker compose, k8s helm etc. We’ve repeatedly optimized the self-host process to enable you to make self-host and software deployments in just a few minutes. 
+Compared to the cloud, you can use ILLA products more securely and reliably, and you can use ILLA Builder by simply visiting the deployment site after hosting and deploying Builder locally.
+
+### Customized Development Workflow Example
+
+Latest version of Node.js LTS and pnpm are installed/ running. Install from links above following instructions.
+
+Clone the illa-builder repository, and run `pnpm install` to fetch its dependencies. Customize your language parameter, i.e. NEXT.js, and create a development workflow environment; input the following to `apps/builder/.env.development.local`
 
 ```
-ILLA_API_BASE_URL=localhost:9345
+ILLA_API_BASE_URL=localhost:<PORT>  # this is your backend address, if this is not present, it will use the default backend address, location.origin, update <PORT> to port number
 ILLA_INSTANCE_ID=SELF_HOST_CLOUD
 ILLA_APP_VERSION=0.0.0
 ILLA_APP_ENV=development
 ILLA_USE_HTTPS=false
 ```
 
 
-#### If you not have rust
+### Deploying Manually
+
+Alternatively, you can deploy Illa Builder manually using Docker Compose and Kubernetes.
+
+Check out instructions for deploying with [Docker Compose](https://github.com/illacloud/deploy-illa-manually/blob/main/docker-compose/README.md), and Instructions for deploying with [Kubernetes](https://github.com/illacloud/deploy-illa-manually/blob/main/kubernetes/README.md).
+
+For more information on self-hosting, please see the ILA CLI, Docker Compose, and k8s helm boot pages. If you have difficulties with your self-deployment, please review the 
+[Deploy Introduction](https://www.illacloud.com/docs/deploy-introduction) Self-host/ Deploy Manually instructions; you can also contact us on GitHub or at business@illasoft.com.
+
+
+## For Rust Developer's
 
-You can use Docker pull image: `docker pull illasoft/illa-builder:latest`, and then you can run `docker run -d -p 9345:2022 illasoft/illa-builder:latest`, this means you will deploy illa-builder-backend at port 9345. And then you can modify your `.env.development.local`
+As a Rust Developer, you can easily build+install the latest ILLA CLI release with cargo, execute command:
 
+`cargo install illa`
 
+The cargo tool will download the ILLA CLI with its source dependencies, build and install it into the cargo bin path so that we can run it. 
 
+Once installed, you can run the ILLA CLI utilizing the illa command:
 
-### If you are go developer
+`./illa <COMMAND>`
+
+Illa CLI is a tool for hosting ILLA Builder in your local environment. Once installed, execute command `./illa deploy --self --port <PORT>`, update `<PORT>` to port number. Again, this will deploy illa-builder-backend at port `<PORT>`. 
+If needing to modify your `.env.self`, input:
```

Comment:
If you use the CLI, don't need to care about this configuration file. This file only used on develop or build

Reply:
Should this specific part on the env.self file be removed, and then a new PR submitted?

Reply:
Hey @kiddjsh 
We will get this PR merged first and our team will keep improving based on your contribution. Thank you for contributing to our Project!!